tests/csapi/room_profile_test.go

```diff
@@ -0,0 +1,47 @@
+package csapi_tests
+
+import (
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+)
+
+func TestAvatarUrlUpdate(t *testing.T) {
+	testProfileFieldUpdate(t, "avatar_url")
+}
+
+func TestDisplayNameUpdate(t *testing.T) {
+	testProfileFieldUpdate(t, "displayname")
+}
+
+// sytest: $datum updates affect room member events
+func testProfileFieldUpdate(t *testing.T, field string) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+
+	const bogusData = "LemurLover"
```

Comment:
For things like `avatar_url` we expect this to be an `mxc://` URI. It would be reasonable to think that a HS impl may reject non-mxc URIs, so we can't put bogus data for this field. Yank the data out to the function args please `testProfileFieldUpdate(t *testing.T, field, val string)`

Reply:
Actually the sytest does this incorrectly so meh, fine.